File: Examples/test-suite/perl5/operator_overload_break_runme.pl
#!/usr/bin/perl -w
use strict;
use Test::More tests => 9;

use operator_overload_break;

pass("loaded");

my $op = operator_overload_break::Op->new(5);

isa_ok($op, "operator_overload_break::Op");

ok((2 == $op - 3),
   "subtraction");

$op->{k} = 37;

ok((40 == $op + 3),
   "addition");

$op->{k} = 22;

ok((10 == (32 - $op)),
   "reversed subtraction");

ok(not(3 == $op),
       'not equal');

$op->{k} = 3;

++$op;

ok(($op == 4),
   "pre-increment operator");

$op++;

ok(($op == 5),
   "post-increment operator");

my $op2 = ++$op;

$op2 = $op++;

ok(($op == 7) and ($op2 == 7),
   "multiple post-increments");

File: Examples/test-suite/lua/operator_overload_break_runme.lua
local v=require("operator_overload_break")

local op = v.Op(5)
assert(op ~= nil)
assert(swig_type(op) == "Op *")

assert(op - 3 == v.Op(2), "subtraction")
op.k = 37
assert(op + 3 == v.Op(40), "addition")
op.k = 22
assert(v.Op(32) - op == v.Op(10), "reversed subtraction")
assert(not (op == v.Op(3)), 'not equal')
assert(op ~= v.Op(3), 'direct not equal')
op.k = 3
--[[ Lua TODO %rename(PlusPlusPrefix) does not work
op:PlusPlusPrefix() -- = (++op)
assert(op == v.Op(4), "pre-increment operator")
op:PlusPlusPostfix() -- = (op++)
assert(op == v.Op(5), "post-increment operator")
local op2 = op:PlusPlusPrefix() -- = (++op)
op2 = op:PlusPlusPostfix() -- = (op++)
assert(op == v.Op(7) and op2 == v.Op(7), "multiple post-increments")
]]
